src/engine/enginebuffer.cpp

```diff
@@ -232,6 +230,18 @@ EngineBuffer::EngineBuffer(const QString& group,
     m_pCueControl = new CueControl(group, pConfig);
     addControl(m_pCueControl);
 
+    for (int i = 1; i <= kMacrosPerChannel; ++i) {
```

Comment:
Creating an instance of a MacroControl for each individual macro is different from how the single CueControl handles all cues for an EngineBuffer. This isn't necessarily a problem. Is this different design intentional?

Reply:
It's not really different, but instantiating the individual hotcuecontrols is hidden in cuecontrol which creates a bunch of hotcuecontrol instances internally. I'd prefer a single instance of macrocontrol which creates a bunch of macroslotcontrols internally to move business logic out of the overcrowded engine buffer class.

Reply:
As already talked over with @Holzhaus, then we would have a class for the sole purpose of running this loop. The big difference in CueControl is that it itself is also a cue.